cmd/fatt/cli/publish.go

```diff
@@ -0,0 +1,102 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/package-url/packageurl-go"
+	"github.com/spf13/cobra"
+
+	"github.com/philips-labs/fatt/cmd/fatt/cli/options"
+	"github.com/philips-labs/fatt/pkg/attestation"
+)
+
+// AttestationsTXT implements the Stringer interface to write the package urls
+// to a newline separated file
+type AttestationsTXT []*packageurl.PackageURL
+
+// Scheme implements remote.File
+func (a AttestationsTXT) Scheme() string {
+	return "discovery"
+}
+
+// Contents implements remote.File
+func (a AttestationsTXT) Contents() ([]byte, error) {
+	var b bytes.Buffer
+	for _, p := range a {
+		b.WriteString(p.String() + "\n")
+	}
+	return b.Bytes(), nil
+}
+
+// Path implements remote.File
+func (AttestationsTXT) Path() string {
+	return "attestations.txt"
+}
+
+// Platform implements remote.File
+func (AttestationsTXT) Platform() *v1.Platform {
+	return nil
```

Comment:
Could you explain a bit what the Platform is and why it currently is implemented and returns nil?

Reply:
It is to satisfy the `github.com/sigstore/cosign/pkg/cosign/remote/File` interface. In oci this is used for e.g. arm vs x64 image architectures. An SBOM or provenance file however isn't platform specific and therefore return nil is a sufficient implementation.

Also see the method documentation.